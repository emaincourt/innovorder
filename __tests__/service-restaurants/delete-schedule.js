import Boom from 'boom';

beforeEach(() => {
  jest.resetModules();
});

describe('Handler', () => {
  it('deletes a schedule according to its id', async () => {
    expect.assertions(3);
    jest.mock('../../sequelize/models', () => ({
      schedule: {
        findById: (id) => {
          expect(id).toEqual('any-schedule-id');
          return Promise.resolve({
            destroy: () => Promise.resolve(),
          });
        },
      },
    }));
    const DeleteSchedule = require('../../services/service-restaurants/delete-schedule').default;
    await expect(
      DeleteSchedule.handler({
        params: {
          scheduleId: 'any-schedule-id',
        },
      }, (response) => {
        expect(response).toEqual({});
      }),
    ).resolves.toMatchSnapshot();
  });

  it('forwards the error if one gets thrown during the deletion', async () => {
    it('deletes a schedule according to its id', async () => {
      expect.assertions(3);
      jest.mock('../../sequelize/models', () => ({
        schedule: {
          findById: (id) => {
            expect(id).toEqual('any-schedule-id');
            return Promise.resolve({
              destroy: () => Promise.reject(new Error()),
            });
          },
        },
      }));
      const DeleteSchedule = require('../../services/service-restaurants/delete-schedule').default;
      await expect(
        DeleteSchedule.handler({
          params: {
            scheduleId: 'any-schedule-id',
          },
        }, (response) => {
          expect(response).toEqual(Boom.wrap(new Error()));
        }),
      ).resolves.toMatchSnapshot();
    });
  });
});
