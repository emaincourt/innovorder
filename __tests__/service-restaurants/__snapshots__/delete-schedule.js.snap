// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler deletes a schedule according to its id 1`] = `undefined`;
